aws-ssm-maintenancewindow/src/main/java/software/amazon/ssm/maintenancewindow/translator/ExceptionTranslator.java

```diff
@@ -0,0 +1,82 @@
+import software.amazon.cloudformation.exceptions.CfnNotFoundException;
+import software.amazon.cloudformation.exceptions.CfnThrottlingException;
+import software.amazon.cloudformation.exceptions.CfnGeneralServiceException;
+import software.amazon.cloudformation.exceptions.CfnInvalidRequestException;
+
+/**
+ * Translator between service model exceptions and CloudFormation exceptions.
+ */
+public class ExceptionTranslator {
+
+    /**
+     * Translates service model exceptions to CloudFormation exceptions.
+     *
+     * @param serviceException Service model exception to translate.
+     * @param request Type of SsmRequest where the service exception came from.
+     * used for better exception messaging for the customer.
+     * @return CloudFormation-type exception converted from a service model exception.
+     */
+    public BaseHandlerException translateFromServiceException(final Exception serviceException,
+                                                              final SsmRequest request) {
+
+        if (serviceException instanceof AlreadyExistsException) {
+
+            return new CfnAlreadyExistsException(serviceException);
+        } else if (serviceException instanceof ResourceLimitExceededException) {
+
+            return new CfnServiceLimitExceededException(serviceException);
+        } else if (serviceException instanceof DoesNotExistException) {
+
+            return new CfnNotFoundException(serviceException);
+        } else if (serviceException instanceof InternalServerErrorException) {
+
+            return new CfnServiceInternalErrorException(
+                    getClassNameWithoutRequestSuffix(request.getClass().getSimpleName()),
+                    serviceException);
+        } else if (serviceException instanceof FeatureNotAvailableException
+                || serviceException instanceof TargetInUseException
+                || serviceException instanceof IdempotentParameterMismatchException){
+
+            return new CfnInvalidRequestException(request.toString(), serviceException);
+        }else if (serviceException instanceof TooManyUpdatesException) {
+
+            return new CfnThrottlingException(getClassNameWithoutRequestSuffix(request.getClass().getSimpleName()),
+                    serviceException);
+        } else {
+            // in case of unknown/unexpected service exceptions, use a generic exception with the name of the failed operation
+            return new CfnGeneralServiceException(getClassNameWithoutRequestSuffix(request.getClass().getSimpleName()),
+                    serviceException);
+        }
+    }
+
+    /**
+     * Removes suffix "Request" string from input class names.
+     *
+     * @param simpleSsmRequestClassName Name of SsmRequest class to remove "Request" suffix from.
+     * @return Request class name without "Request" in the end.
+     */
+    private String getClassNameWithoutRequestSuffix(final String simpleSsmRequestClassName) {
```

Comment:
Might be worth moving this over to a shared class later, as we also use this in AWS::SSM::Association.

Reply:
Yeah agreed but that can be done separately, i would prefer that done outside this pr is its not a hard blocker you believe.